packages/api/core/src/util/require-search.ts

```diff
@@ -14,7 +14,13 @@ export function requireSearchRaw<T>(relativeTo: string, paths: string[]): T | nu
       // eslint-disable-next-line global-require, import/no-dynamic-require
       return require(testPath);
     } catch (err) {
-      // Ignore the error
+      // If the error is a require error, and it's for the file we're looking for - ignore it
+      if (err.code === 'MODULE_NOT_FOUND' && (err.requestPath === undefined || err.requestPath === testPath)) {
+        // Ignore the error
+      } else {
+        // Re-throw the error
+        throw err;
+      }
```

Comment:
We want to avoid empty conditional blocks when possible.
```suggestion
      // Ignore require-related errors
      if (err.code !== 'MODULE_NOT_FOUND' || ![undefined, testPath].includes(err.requestPath)) {
        throw err;
      }
```

Although it would be a little more ideal if there was a specific Error subclass to check for.

Reply:
> Although it would be a little more ideal if there was a specific Error subclass to check for.

Sorry, that's not possible.

https://nodejs.org/api/modules.html#modules_file_modules

> If the given path does not exist, require() will throw an `Error` with its code property set to 'MODULE_NOT_FOUND'.

Reply:
> Sorry, that's not possible.

That's unfortunate.